core/src/main/scala/tech/beshu/ror/accesscontrol/factory/JsonConfigStaticVariableResolver.scala

```diff
@@ -54,17 +54,31 @@ object JsonConfigStaticVariableResolver {
       .mapObject(_.mapValues(mapJson(_, errors)))
       .withString { str =>
         val resolved = NonEmptyString.unapply(str) match {
-          case Some(nes) => tryToResolveAllStaticSingleVars(nes, errors)
+          case Some(nes) =>
+            tryToResolveAllStaticSingleVars(nes, errors)
           case None => str
         }
-        resolvedStringToJson(resolved)
+        if (resolved =!= str)
+          resolvedStringToJson(resolved, json)
+        else json
       }
   }
 
-  private def resolvedStringToJson(resolvedStr: String) = {
+  private def preserveNumbersAsStrings(original: Json, newValue: Json) = {
```

Comment:
the method is used only by the `resolvedStringToJson` method, so it should be defined after it (for a sake of readability)